openhands/llm/llm.py

```diff
@@ -240,12 +245,20 @@ def wrapper(*args, **kwargs):
                     resp.choices[0].message = fn_call_response_message
 
                 message_back: str = resp['choices'][0]['message']['content'] or ''
-                tool_calls = resp['choices'][0]['message'].get('tool_calls', [])
+                tool_calls: list[ChatCompletionMessageToolCall] = resp['choices'][0][
+                    'message'
+                ].get('tool_calls', [])
                 if tool_calls:
                     for tool_call in tool_calls:
-                        fn_name = tool_call.function.name
-                        fn_args = tool_call.function.arguments
-                        message_back += f'\nFunction call: {fn_name}({fn_args})'
+                        fn: Function | None = (
+                            tool_call.model_extra.get('function', None)
```

Comment:
This this the best practice to access function from `model_extra`?

Reply:
Quite the contrary, I'd rather not rely on `model_extra` at all. Even though we do this for other things. The reason for this change is that it wasn't logging at all, getting function didn't work. Which is weird... Let me try again.

![image](https://github.com/user-attachments/assets/aa270389-ae8b-4dc0-9af7-ad6d4bb66a3f)

Reply:
Reverted. litellm aims to be openai-compatible so it uses some magic to do it - and it works for me now